### PR TITLE
Rewrite the MobileBuyIntegration sample controllers in SwiftUI

### PR DESCRIPTION
**Pull request by markmur**
### What changes are you making?

- [x] Rewrite the MobileBuyIntegration sample controllers in SwiftUI
- [x] Update branding

https://github.com/user-attachments/assets/87e0eb81-f71d-4c6d-a27e-8c5d0c1c6138

---

### Before you merge

> [!IMPORTANT]
>
> - [x] I have read and agree with the [Contribution Guidelines](https://github.com/shopify/checkout-sheet-kit-swift/blob/main/.github/CONTRIBUTING.md).
> - [x] I have read and agree with the [Code of Conduct](https://github.com/shopify/checkout-sheet-kit-swift/blob/main/.github/CODE_OF_CONDUCT.md).
  > - [ ] I've added tests to support my implementation
> - [ ] I've updated the [README](https://github.com/shopify/checkout-sheet-kit-swift).

---

### CODE REVIEW

File: Samples/MobileBuyIntegration/MobileBuyIntegration/MoneyV2+Format.swift [hunk 1]
```diff
@@ -29,6 +29,10 @@ extension Storefront.MoneyV2 {
 		let formatter = NumberFormatter()
 		formatter.numberStyle = .currency
 		formatter.currencyCode = currencyCode.rawValue
-		return formatter.string(from: NSDecimalNumber(decimal: amount))
+		return isFree() ? "Free" : formatter.string(from: NSDecimalNumber(decimal: amount))
```
Comment (kiftio) [inline Samples/MobileBuyIntegration/MobileBuyIntegration/MoneyV2+Format.swift:32]: Gonna steal this for android 👍

File: Samples/MobileBuyIntegration/MobileBuyIntegration/ProductView.swift [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+import UIKit
+import SwiftUI
+import ShopifyCheckoutSheetKit
+
+struct ProductView: View {
+    // MARK: Properties
+    @State private var product: Storefront.Product
+    @State private var handle: String?
+    @State private var loading = false
+	@State private var imageLoaded: Bool = false
```
Comment (kiftio) [inline Samples/MobileBuyIntegration/MobileBuyIntegration/ProductView.swift:34]: Can't tell if it's Github or the indentation is actually a bit off?

File: Samples/MobileBuyIntegration/MobileBuyIntegration/ProductView.swift [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+						.padding(.vertical)
+                        .foregroundColor(Color(ColorPalette.primaryColor))
+
+					Text(product.title)
+                        .font(.title)
+
+                    Text(product.description)
+                        .font(.body)
+                        .foregroundColor(.gray)
+                        .lineLimit(descriptionExpanded ? 10 : 3)
```
Comment (kiftio) [inline Samples/MobileBuyIntegration/MobileBuyIntegration/ProductView.swift:96]: I think 10 for the expanded text is still bit too low for some of the products in the store (e.g. looking at Bird of paradise)

File: Samples/MobileBuyIntegration/MobileBuyIntegration/ProductView.swift [hunk 4]
```diff
@@ -0,0 +1,290 @@
...
+        }
+		.navigationTitle(product.collections.nodes.first?.title ?? product.title)
+    }
+
+    // MARK: Methods
+    private func addToCart() {
+        guard let variant = product.variants.nodes.first else { return }
+
+        loading = true
+        let start = Date()
```
Comment (kiftio) [inline Samples/MobileBuyIntegration/MobileBuyIntegration/ProductView.swift:135]: Continuous clock looks interesting.. Although I'm not sure how easy it is to use with callbacks.

```
let clock = ContinuousClock()
let elapsed = clock.measure {
    someWork()
}
```

File: Samples/MobileBuyIntegration/MobileBuyIntegration/SettingsViewController.swift [hunk 5]
```diff
@@ -42,14 +41,14 @@ struct SettingsView: View {
...
-					Toggle("Prefill buyer information", isOn: $appConfiguration.useVaultedState)
+					Toggle("Prefill buyer information", isOn: $config.useVaultedState)
 				}
 
 				Section(header: Text("Universal Links")) {
-					Toggle("Handle Checkout URLs", isOn: $appConfiguration.universalLinks.checkout)
-					Toggle("Handle Cart URLs", isOn: $appConfiguration.universalLinks.cart)
-					Toggle("Handle Product URLs", isOn: $appConfiguration.universalLinks.products)
-					Toggle("Handle all Universal Links", isOn: $appConfiguration.universalLinks.handleAllURLsInApp)
+					Toggle("Handle Checkout URLs", isOn: $config.universalLinks.checkout)
```
Comment (kiftio) [inline Samples/MobileBuyIntegration/MobileBuyIntegration/SettingsViewController.swift:48]: We also need to catch android up here at some point

**Review by kiftio — APPROVED** [review]

File: Samples/MobileBuyIntegration/MobileBuyIntegration/ProductView.swift (hunk 2 above)
Comment (markmur) [reply Samples/MobileBuyIntegration/MobileBuyIntegration/ProductView.swift:34]: Yeah it's actually off. https://github.com/Shopify/checkout-sheet-kit-swift/pull/247 will fix it

File: Samples/MobileBuyIntegration/MobileBuyIntegration/ProductView.swift (hunk 3 above)
Comment (markmur) [reply Samples/MobileBuyIntegration/MobileBuyIntegration/ProductView.swift:96]: Yeah, probably. I'll account for this in the full redesign task

**Pull request merged**